hedera-node/src/main/java/com/hedera/services/ledger/BalanceChange.java

```diff
@@ -0,0 +1,143 @@
+package com.hedera.services.ledger;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.google.common.base.MoreObjects;
+import com.hedera.services.store.models.Id;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.TokenID;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INSUFFICIENT_ACCOUNT_BALANCE;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INSUFFICIENT_TOKEN_BALANCE;
+
+/**
+ * Process object that encapsulates a balance change, either ℏ or token unit.
+ *
+ * Includes an optional override for the {@link ResponseCodeEnum} to be used
+ * in the case that the change is determined to result to an insufficient balance;
+ * and a field to contain the new balance that will result from the change.
+ * (This field is helpful to simplify work done in {@link HederaLedger}.)
+ *
+ * The {@code explicitTokenId} and {@code explicitAccountId} fields are
+ * temporary, needed to interact with the {@link com.hedera.services.ledger.accounts.BackingAccounts}
+ * and {@link com.hedera.services.ledger.accounts.BackingTokenRels} components
+ * whose APIs still use gRPC types.
+ */
+public class BalanceChange {
+	private final Id token;
+	private final Id account;
+	private final long units;
+	private ResponseCodeEnum codeForInsufficientBalance = INSUFFICIENT_ACCOUNT_BALANCE;
+
+	private long newBalance;
```

Comment:
What's the relationship between `units` and `newBalance`?

Reply:
The `units` field is the **change** to be made to a balance.

The `newBalance` field is just a convenient place for downstream code to put the result of adding `units` to the existing balance.